### DISCUSSION TITLE
Under what conditions does `finishedOn` property is null

### DISCUSSION THREAD

**Discussion by pmadhavan** [taskforcesh/bullmq#603] [Q&A]
I am trying to track metrics around time taken to process a job or duration of the job staring from when it was picked to when it was completed. I use the properties `finishedOn` and `processedOn` to calculate the same (`job.finishedOn - job.processedOn`), but most of the time the finishedOn property returns null.

Since the property is nullable, I am doing something like `job.finishedOn || Date.now() - job.processedOn` as a workaround in the `onWorkerComplete` method.

Job should either be completed or failed, curious to know why would `finishedOn` property would be null?

**Comment by manast** [comment]
```finishedOn``` is set when a job completes or fails, so if that if the case it cannot be null unless there is a bug somewhere.
